Sources/eppo/dto/EppoValue.swift

```diff
@@ -2,154 +2,167 @@ import Foundation
             self.type = .ArrayOfStrings
-            self.array = array
-            self.value = nil
-        } else if let numericValue = try? container.decode(Double.self) {
+            self.stringArrayValue = array
+        } else if let doubleValue = try? container.decode(Double.self) {
             // decode double handles both integers and floating-point numbers.
-            //
-            // todo: this class is clunky with storing ints and doubles as strings.
-            // add support for storing them natively to avoid repeated conversions.
-            self.type = .Number
-            self.value = String(numericValue)
+            self.type = .Numeric
+            self.doubleValue = doubleValue
         } else if let boolValue = try? container.decode(Bool.self) {
             self.type = .Boolean
-            self.value = String(boolValue)
+            self.boolValue = boolValue
         } else if let stringValue = try? container.decode(String.self) {
-                self.type = .String
-                self.value = stringValue
-            } else {
+            self.type = .String
+            self.stringValue = stringValue
+        } else {
             self.type = .Null
-            self.value = nil
         }
     }
 
-    public static func valueOf() -> EppoValue {
-        return EppoValue(type: EppoValueType.Null);
-    }
-
     public static func valueOf(_ value: Bool) -> EppoValue {
-        return EppoValue(value: String(value), type: EppoValueType.Boolean);
+        return EppoValue(value: value);
     }
-    
+
     public static func valueOf(_ value: Double) -> EppoValue {
-        return EppoValue(value: String(value), type: EppoValueType.Number);
+        return EppoValue(value: value);
     }
     
+    public static func valueOf(_ value: Int) -> EppoValue {
+        return EppoValue(value: value);
+    }
+
     public static func valueOf(_ value: String) -> EppoValue {
-        return EppoValue(value: value, type: EppoValueType.String);
+        return EppoValue(value: value);
     }
 
     public static func valueOf(_ value: [String]) -> EppoValue {
         return EppoValue(array: value);
     }
+    
+    public func isNull() -> Bool {
+        return self.type == EppoValueType.Null
+    }
+    
+    public func isBool() -> Bool {
+        return self.type == EppoValueType.Boolean
+    }
+    
+    public func isNumeric() -> Bool {
+        return self.type == EppoValueType.Numeric
+    }
 
-    public func boolValue() throws -> Bool {
-        if self.value == nil {
-            throw Errors.valueNotSet;
-        }
-
-        guard let rval = Bool(self.value!) else {
-            throw Errors.conversionError;
-        }
-
-        return rval;
+    public func isString() -> Bool {
+        return self.type == EppoValueType.String
     }
     
-    public func doubleValue() throws -> Double {
-        if self.value == nil {
-            throw Errors.valueNotSet;
+    public func getBoolValue() throws -> Bool {
```

Comment:
Leaving this as a `throws` - alternatively can return a `Bool?`. Either way a missing value needs to be handled

Reply:
Yeah I think missing values should throw